docs/blueprints/EP038.rst

```diff
@@ -0,0 +1,90 @@
+
+  - Adding **"in_port"** as matching criteria to the created flow. **coloring** recognizes only discovered NNIs. Their ports can be utilize to specify the matching so there is going to be one flow per NNI port. Examples flows 1 and 2.
+  - Add an extra flow to drop the unwanted traffic. It will match any packet with MAC prefix used by coloring. It can be done using mask. Example flow 3.
+
+Diagram on how the solution would look like:
+
+  .. image:: https://drive.google.com/uc?export=view&id=1Wmlr-u4Tgw4YWNe46M2CZcRvXk_MCnlZ
+
+Priority
+--------
+
+This solution is not going to stop loads of packets comming from not NNIs but just manage them. In order to free the rest of flows, some priorities need to be adjusted:
+
+- Leave the priority **50 000** for the **coloring** flow entries with **"in_port"** match in them
+- The flow entries that drop packet should have a higher priority than **mef_eline** entries (max. **20 000**) and lower or equal than **of_lldp** entries (currently as of writting **50 000**)
+
+Final result for switch 1 with an intra-switch EVC could look like:
+
+  .. code:: console
+
+    cookie=0xac, table=0, send_flow_rem priority=50000,in_port="s1-eth3",dl_src=ee:ee:ee:ee:ee:02 actions=CONTROLLER:65535
+    cookie=0xab, table=0, send_flow_rem priority=50000,dl_vlan=3799,dl_type=0x88cc actions=CONTROLLER:65535
+    cookie=0xac, table=0, send_flow_rem priority=30000,dl_src=ee:ee:ee:00:00:00/ff:ff:ff:00:00:00 actions=drop
+    cookie=0xaa, table=0, send_flow_rem priority=20000,in_port="s1-eth1",dl_vlan=100 actions=set_field:4196->vlan_vid,output:"s1-eth2"
+    cookie=0xaa, table=0, send_flow_rem priority=20000,in_port="s1-eth2",dl_vlan=100 actions=set_field:4196->vlan_vid,output:"s1-eth1"
+
+Settings
+--------
+
+DL_SRC
+~~~~~~
+
+The matching field in **coloring** was configurable from **settings.py** but it is not neccessary. We are going to use the combination of **in_port** and **dl_src** when matching color packets and only **dl_src** to drop not recognize ones. **dl_src** will have a default value **ee:ee:ee:00:00:00/ff:ff:ff:00:00:00** that matches currently used colors (no more than 32 per switch). This way we can reduce the total number of flows.
+
+Since we are going to have a set value for **dl_src**, we can make it configurable as **DL_MASK** to reduce its range for small networks or increase it.
+
+- **DL_MASK = "ee:ee:ee:00:00:00/ff:ff:ff:00:00:00"**
+
+Priority
+~~~~~~~~
+
+Adding **PRIORITY_MATCH** and **PRIORITY_DROP**, these are related to priority for matching flows and dropping flows respectively. As default values:
+
+- **PRIORITY_MATCH = 50000**
+- **PRIORITY_DROP = 40000**
+
+**PRIORITY_DROP** should always be lower than **PRIORITY_MATCH** otherwise every color packet will be dropped. Also, it should be higher than **mef_eline** so it frees the succeeding flows.
```

Comment:
@Alopalao please add an additional section on this blueprint to cover other implementation details that will need to be delivered after this blue print. For instance:

- should we provide a migration script to take care of the old flows? 
- what should happen if the operator changes  the DL_MASK on settings? will we send a DELETE all flows with coloring's cookie ID prefix at the begin? (when switch connects?)
- suppose that you want to disable this protection, how should the operator proceed to disable this protection? (maybe something with the PRIORITY_DROP to not install the drop flow -- 0?)

Reply:
maybe this new section could be called "Points of attention" or similar.

Just got a new one:

- The color-to-field method on coloring should be adapted to comply with the new dl_src prefix